src/shared/components/case-editor/services/cases.service.ts

```diff
@@ -87,6 +91,40 @@ export class CasesService {
       );
   }
 
+  /**
+   * handleNestedDynamicListsInComplexTypes()
+   * Reassigns list_item and value data to DymanicList children
+   * down the tree. Server response returns data only in
+   * the `value` object of parent complex type
+   *
+   * EUI-2530 Dynamic Lists for Elements in a Complex Type
+   *
+   * @param jsonResponse - {}
+   */
+  private handleNestedDynamicListsInComplexTypes(jsonResponse) {
+
+    if (jsonResponse.case_fields) {
+      jsonResponse.case_fields.forEach(caseField => {
+        if (caseField.field_type && caseField.field_type.type === CasesService.SERVER_RESPONSE_FIELD_TYPE_COMPLEX) {
+
+          caseField.field_type.complex_fields.forEach(field => {
```

Comment:
I don't know the exact data structure, but this is a loop within a loop. Could you not have done some recursion here instead?

Reply:
You are probably right, but I will need to revisit this code to fix collections anyway. I'll figure it out then.